### PR TITLE
Add callbacks for cellMusic

### PR DESCRIPTION
**Pull request by scribam**
Samples games to test:
- [X] .hack//Versus (check for regressions)
- [X] Astro Tripper (Help & Options > Settings > Custom Music)
- [X] Fuel Overdose (Options > Game Settings > Custom Playlist)
- [X] J-STARS Victory VS+ (Options > BGM Editor)
- [X] JoJo's Bizarre Adventure: All Star Battle (check for regressions)
- [X] Macross 30: The Voice that Connects the Galaxy (check for regressions)
- [X] My Aquarium (Music > HDD)
- [X] Pain (Extras > Jukebox)
- [X] **Tekken Tag Tournament 2 (Ingame)**
- [X] Savage Moon (Options > Audio Options > Use Custom Soundtrack)

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellMusic.cpp [hunk 1]
```diff
@@ -2,131 +2,287 @@
...
+	cellMusic.todo("cellMusicSetSelectionContext2(context=*0x%x)", context);
+
+	const auto music = fxm::get_always<music_t>();
+
+	if (!music->func)
+		return CELL_MUSIC2_ERROR_GENERIC;
+
+	sysutil_register_cb([=](ppu_thread& ppu) -> s32
+	{
+		music->func(ppu, CELL_MUSIC2_EVENT_SET_SELECTION_CONTEXT_RESULT, vm::addr_t(CELL_OK), music->userData);
```
Comment (scribam) [inline rpcs3/Emu/Cell/Modules/cellMusic.cpp]: I am not sure if `vm::addr_t(CELL_OK)` is the right thing to do here. `vm::null` could also be a solution.
This value can be casted to an `uint32_t` inside the callback before usage.

Comment (toccata10) [reply rpcs3/Emu/Cell/Modules/cellMusic.cpp]: Jojo is still working fine, although I've got some freezes from time to time (memory access).
TekkenTagTournament2 goes ingame and can be played, but it has more glitches (no bgm for example). Freezes randomly too.
The 2 log files:
[RPCS3.log.JoJo2.gz](https://github.com/RPCS3/rpcs3/files/1744993/RPCS3.log.JoJo2.gz)
[RPCS3.log-Tekken2.gz](https://github.com/RPCS3/rpcs3/files/1744995/RPCS3.log-Tekken2.gz)
Nice to see some progress for these games.

**Comment by spyropt** [comment]
I 'm trying to run tekken tag 2 but it always fails with a trap
[RPCS3.log.gz](https://github.com/RPCS3/rpcs3/files/1754226/RPCS3.log.gz)

**Comment by spyropt** [comment]
update the problem was on my end tested on another pc and its ok, after creating a new installation the trap disappeared

**Pull request merged**